addons/mail/static/src/core/web/mail_composer_attachment_list.js

```diff
@@ -1,11 +1,27 @@
 import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
 import {
     many2ManyBinaryField,
     Many2ManyBinaryField
 } from "@web/views/fields/many2many_binary/many2many_binary_field";
 
 export class MailComposerAttachmentList extends Many2ManyBinaryField {
     static template = "mail.MailComposerAttachmentList";
+    /** @override */
+    setup() {
+        super.setup();
+        this.mailStore = useService("mail.store");
+        this.attachmentUploadService = useService("mail.attachment_upload");
+    }
+    /**
+     * @override
+     * @param {integer} fileId
+     */
+    async onFileRemove(fileId) {
+        super.onFileRemove(fileId);
+        const attachment = this.mailStore.Attachment.get(fileId);
+        await this.attachmentUploadService.unlink(attachment);
```

Comment:
Indeed not a good idea to rely on `Attachment.get()`, this assumes the data was fetched or inserted prior, so there's implicit coupling with some other code that is not local.

I think using `.get()` is a bad smell and it was kind of a mistake to have, probably we should remove it when we removed all occurrences at some point. Here the best thing to do is assume it might not exist, e.g. `Attachment.insert(fileId)` (yes there's a syntactic sugar when there's only 1 id, we can pass immediately the id value rather than `{ id: fileId }`. The `insert()` will add it in store if it's missing, otherwise it will just retrieve the existing attachment.